pkg/api/globalquery/client/client.go

```diff
@@ -59,3 +90,50 @@ func (c *Client) Query(ctx context.Context, args *query.Args) (*results.Result,
 
 	return res, nil
 }
+
+// Query performs the global query and returns its result, while consuming updates to partial results
+func (sse *SSEClient) Query(ctx context.Context, args *query.Args) (*results.Result, error) {
+	logger := logging.FromContext(ctx)
+
+	if sse.onFinish == nil || sse.onUpdate == nil {
+		return nil, errors.New("no event callbacks provided (onUpdate, onFinish)")
+	}
+
+	// use a copy of the arguments, since some fields are modified by the client
+	queryArgs := *args
+
+	// whatever happens, the results are expected to be returned in json
+	queryArgs.Format = "json"
+
+	if queryArgs.Caller == "" {
+		queryArgs.Caller = clientName
+	}
+
+	buf := &bytes.Buffer{}
+
+	err := jsoniter.NewEncoder(buf).Encode(queryArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Info("calling SSE route")
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, sse.NewURL(api.SSEQueryRoute), buf)
```

Comment:
Mmmmh, should we maybe add this on the `httpc` layer of things so it can be used globally?